doc/jenkins.rst

```diff
@@ -0,0 +1,138 @@
+from ``<main domain>`` to ``<resource domain>`` (e.g. ``my.jenkins.io`` to ``res.my.jenkins.io``).
+
+Note that Jenkins interaction with resource domain, resolved to the same host IP is not done via ``localhost``
+network interface. The reason for that is Jenkins requiring some interaction with itself via this domain name.
+This means that both ``<main domain>`` and ``<resource domain>`` domain names must be resolved correctly, either
+globally (via DNS) or locally on both client and server machines (via ``/etc/hosts`` files). The correctness of
+name resolving is checked when saving the changes to this setting; but Jenkins will only show warning, and not
+fail if domain name is not resolved.
+
+.. note::
+
+    If main server domain name is not resolved using DNS, ``/etc/hosts`` or any other means, the web-interface
+    will only be accessible via IP, and not the name. Because of that, the Jenkins interaction with itself
+    via domain name will fail because host name is not passed to the Jenkins server
+
+Here are the symptoms of domain names not resolving correctly:
+
+1. Jenkins warnings when trying to save the updated settings
+2. Client inability to access said pages (timeout error)
+
+To set up domain name resolving, add following lines to server ``/ets/host`` file::
+
+    127.0.0.1       <main domain>
+    <server IP>     <resource domain>
+
+And add the following lines to client ``/ets/host`` file::
+
+    <server IP>     <main domain>
+    <server IP>     <resource domain>
+
+
+.. _jenkins#nginx:
+
+Configure reverse proxy
+-----------------------
+
+Note that this step is only needed if `Nginx reverse proxy
+<https://docs.nginx.com/nginx/admin-guide/web-server/reverse-proxy/>`__ is used.
+
+To understand why these fixes are needed, let's pay more attention to the mechanism of 'another domain', used by
+Jenkins. When requesting an artifact, that is served from another domain, user first goes to main Jenkins web
+server, that returns a redirection link to acquire a said artifact.
+
+As `specified in docs <https://docs.nginx.com/nginx/admin-guide/web-server/reverse-proxy/#passing-request-headers>`__,
+without specification Nginx replaces ``Host`` headers with ``$proxy_host``. In this case it changes
+``res.my.jenkins.io`` to proxy IP and port specifications. The problem is, when received such specifications,
+Jenkins server no longer understands the request is sent to resource domain, and returns the ``404 NOT FOUND`` error.
+
+To pass them correctly, adjust the configuration as instructed in manual mentioned above. Add the following lines
+to Nginx configuration file::
+
+    location / {
+        proxy_set_header Host $host;
+        proxy_set_header X-Real-IP $remote_addr;
+    }
+
+so that real headers are passed to Jenkins to handle the resource domain magic.
+
+.. note::
+
+    Also you might need to set ``server_name`` to ``my.jenkins.io res.my.jenkins.io`` if they are located
```

Comment:
The difference between one or two server entries is is defined by the difference of their external IP addresses.
If the physical server is the same and the external IP address is the same, both can be described by a single entry within nginx config.

If there are two different IPs, two options are possible:
1. they are still hosted on the same machine -> a single nginx instance can server the (single config file), but two separate entries are required
2. they are hosted on two different machines -> two nginx instances are requied -> two separate config files

I propose to remove "same host" and related parts from this sentence.